### PR TITLE
GH-1094 Extended deployment domain service for multiple wallets

### PR DESCRIPTION
**Pull request by DmitryBespalov**
- Refactored methods to strip out boilerplate helpers code
- Added walletID to each wallet domain event
- Deployment service will start deploying all wallets that are in creation states
- Unit tests got review and refactoring, too.
- New repository method to filter wallets by state

Roadmap:
- [x] Adjust `DeploymentDomainService` and creation events to use explicit `walletID` instead of `selectedWallet`
- [ ] Adjust flow coordinators: clean up draft safes on startup, before new safe starts, and after create safe flow exits (for any reason)
- [ ] Add menu item to create safe; add "Menu" button to the "no funds" screens.
- [ ] Check that menu is updating on safe status updates
- [ ] On app start: resume deployment service

### CODE REVIEW

File: MultisigWallet/MultisigWalletDomainModel/Wallet/DeploymentDomainService.swift [hunk 1]
```diff
@@ -83,172 +57,132 @@ public class DeploymentDomainService {
...
-            } else {
-                let userInfo: [String: Any] = [NSLocalizedDescriptionKey: "Creation transaction failed"]
-                let error = NSError(domain: "DeploymentDomainService", code: -1, userInfo: userInfo)
-                DomainRegistry.logger.error("Wallet creation transaction failed", error: error)
-                wallet.cancel()
-            }
-        }
+    func crashTheApp(_ wallet: Wallet) throws {
+        // should we still crash the app if the wallet failed? probably not.
+        // what then? then we should notify the user of it and ask to contact us.
```
Comment (sche) [inline MultisigWallet/MultisigWalletDomainModel/Wallet/DeploymentDomainService.swift:159]: Should we add a TODO here? It is easy to forget

File: MultisigWallet/MultisigWalletDomainModel/Wallet/DeploymentDomainService.swift [hunk 2]
```diff
@@ -83,172 +57,132 @@ public class DeploymentDomainService {
...
+
+    /// Assigns a handler to a wallet event of certain type
+    private func subscribe<T: WalletEvent>(event: T.Type, action: @escaping (Wallet) throws -> Void) {
+        DomainRegistry.eventPublisher.subscribe(self, bind(event, action))
     }
 
-    func creationFailed(_ event: WalletCreationFailed) {
-        DomainRegistry.system.exit(EXIT_FAILURE)
+    /// Wrapper for a wallet event handler
+    private func bind<T: WalletEvent>(_ event: T.Type, _ action: @escaping (Wallet) throws -> Void) -> (T) -> Void {
```
Comment (sche) [inline MultisigWallet/MultisigWalletDomainModel/Wallet/DeploymentDomainService.swift:175]: OMG, blowing up my mind... It works, but it is too complex syntax imo

**Review by sche — APPROVED** [review]

File: MultisigWallet/MultisigWalletDomainModel/Wallet/DeploymentDomainService.swift (hunk 1 above)
Comment (DmitryBespalov) [reply MultisigWallet/MultisigWalletDomainModel/Wallet/DeploymentDomainService.swift:159]: I have it on my list

File: MultisigWallet/MultisigWalletDomainModel/Wallet/DeploymentDomainService.swift (hunk 2 above)
Comment (DmitryBespalov) [reply MultisigWallet/MultisigWalletDomainModel/Wallet/DeploymentDomainService.swift:175]: 💥

**Pull request merged**